LPS-177199 tests to cover virtual instance migrator validator

lucasperj opened the pull request:
Previous PR: https://github.com/liferay-uniform/liferay-portal/pull/1324
---
Story : https://liferay.atlassian.net/browse/LPS-171386
Automation ticket : https://liferay.atlassian.net/browse/LPS-177199

cc/ @vicnate5

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upgrade**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-26.liferay.com/job/test-portal-source-format/6485/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-source-format/builds/6485/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-26.liferay.com/job/test-portal-source-format/6485/">test-portal-source-format#6485</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-source-format/builds/6485/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-uniform/liferay-portal/pull/1338">liferay-uniform#1338</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2406144847">[master] ci:test:sf - lucasperj &gt; liferay-uniform - PR#1338 - 2023-08-14[13:03:23]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-26.liferay.com/job/publish-testray-report/25058/">publish-testray-report#25058</a><br/></div></div>

vicnate5 commented:
@lucasperj I would still like to see some more refinement here and I have some questions:

## 1

For the catalina opts, append it rather than full replace (since you aren't changing the defaults)

I would just add another entry, so the file looks like:
```
CATALINA_OPTS="$CATALINA_OPTS -Dfile.encoding=UTF-8 -Djava.locale.providers=JRE,COMPAT,CLDR -Djava.net.preferIPv4Stack=true -Duser.timezone=GMT -Xms2560m -Xmx2560m -XX:MaxNewSize=1536m -XX:MaxMetaspaceSize=768m -XX:MetaspaceSize=768m -XX:NewSize=1536m -XX:SurvivorRatio=7"

CATALINA_OPTS="$CATALINA_OPTS -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=1234 -Dcom.sun.management.jmxremote.ssl=false"
```
See also build-test.xml `setup-testable-tomcat-jacoco-agent`

## 2
Please finish applying the given-when-then pattern as mentioned in https://github.com/liferay-uniform/liferay-portal/pull/1324#issuecomment-1671854539. The final task should usually be a `Then`.

## 3

I noticed that all the tests are running this at the beginning `AntCommands.runCommand("modules/util/portal-tools-db-upgrade-client/src/testFunctional/ant/build-test-db-upgrade-client.xml", "check-mbean-attributes");`

Can this be triggered by a property instead? We could get the testcase property, if true, then generate the output right after startup.

## 4

What happened to the `modify-setenv-script` needed in the previous iteration? Is it no longer needed for some reason?

## 5

I am curious how `CheckMbeanDuringUpgradeIsRunning` and `CheckMbeanFailedUpgrade`  work. I would expect, given that the assertions are in poshi, wouldn't the upgrade have completed or failed out before the Poshi assertions can be done?

lucasperj commented:
@vicnate5 , 

1. I chose to replace the entire catalina opts so that later, after running the test, I could just replace it exactly as it was before, since I'm capturing the original state in a property.
2. ok , they are finishing with a "And" which follows from a then, but I'll condense
3. I believe can be done, applying the logic to the end startup target/macro maybe, I just don't think it's going to be as clear as having your logic defined in a file destined for the feature.
4. _What happened to the modify-setenv-script needed in the previous iteration? Is it no longer needed for some reason?_
Now it is applied before starting the test as shown in [c0630866432e7f342d4c52b41093bd9fe8b54a28](https://github.com/liferay-uniform/liferay-portal/pull/1338/commits/c0630866432e7f342d4c52b41093bd9fe8b54a28) , this macro was just to change the catalina_opts, as well as the logic for clean up changes after test is over
5. _I am curious how CheckMbeanDuringUpgradeIsRunning and CheckMbeanFailedUpgrade work. I would expect, given that the assertions are in poshi, wouldn't the upgrade have completed or failed out before the Poshi assertions can be done?_

For CheckMbeanDuringUpgradeIsRunning, it falls under the condition that if the property is active, it will capture the mBean attributes in parallel with the upgrade running

> <parallel>
					<start-app-server />

					<sequential>
						<wait-for-app-server-log-message log.message="Started web bundles" />

						<get-mbean-attributes />
					</sequential>
				</parallel>

, so it manages to get the attribute as running, when it captures it it already throws it to the file with the output.
For CheckMbeanFailedUpgrade, as we use auto.upgrade, it doesn't force the connection to gogo-shell in case of failure, it captures the attribute after the upgrade has completed.
In **both** cases, the assertion in the output file only occurs after the Upgrade is finished, but the capture of attributes in the case of the first test, is during, so the consistency

vicnate5 commented:
@lucasperj For 1, I still think it is better to append with the new line (expanded catalina opts) and then just delete the line afterwards.

For 3, call the property, `upgrade.jmx.mbean.output.enabled`. Set it to true and generate the output file. Test can then assume the file is there.

Thanks for the explanations of the other ones.

lucasperj commented:
ci:test:upgrade

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>upgrade</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest(master)/7846/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-29.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7846/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:upgrade - 0 out of 1 jobs passed in 2 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/9c4968c530400f38c780380f006d32d1aa664e35">9c4968c530400f38c780380f006d32d1aa664e35</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/9c4968c530400f38c780380f006d32d1aa664e35">9c4968c530400f38c780380f006d32d1aa664e35</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2405917790">EE Development Acceptance (master) - 4739 - 2023-08-13[20:45:33]</a></p><details><summary><strong>ci:test:upgrade - 0 out of 1 jobs PASSED</strong></summary><div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest(master)/7846/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li></ul></div></details><h5>For more details click <a href="https://test-1-29.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7846/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest(master)/7846/">test-portal-acceptance-pullrequest(master)</a></h5><div><h6>Job Results:</h6><p>0 Jobs Passed.<br/>1 Job Failed.</p></div><div><p>The sender branch SHA could not be found on <strong><a href="https://github.com/liferay-uniform/liferay-portal/pull/1338">lucasperj/LPS-177199</a></strong>. The sender branch may have been force pushed or deleted after the pull request test was initiated.</p><pre>fatal: Not a valid branch point: '6dfcaee771f8a312c5e653cb24158a7e5327145d'.</pre></div></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-29/jobs/test-portal-acceptance-pullrequest(master)/builds/7846/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-29/jobs/test-portal-acceptance-pullrequest(master)/builds/7846/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-29/jobs/test-portal-acceptance-pullrequest(master)/builds/7846/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>upgrade</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/4476/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/4476/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:upgrade - 73 out of 130 jobs passed in 12 hours</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/2a91798016e3edcb4265da3039ad3157ba7ded7c">2a91798016e3edcb4265da3039ad3157ba7ded7c</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/2a91798016e3edcb4265da3039ad3157ba7ded7c">2a91798016e3edcb4265da3039ad3157ba7ded7c</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2406325252">EE Development Acceptance (master) - 4740 - 2023-08-14[20:37:13]</a></p><details><summary><strong>ci:test:upgrade - 73 out of 130 jobs PASSED</strong></summary><div><h4>57 Failed Jobs:</h4><ul><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/4476/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li><li>functional-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/238641/">functional-tomcat90-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/246599/">functional-tomcat90-mysql57-jdk8/0/1</a></li></ul></li><li>modules-integration-mysql57-jdk8<ul><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639371/"><strike>modules-integration-mysql57-jdk8/0/0</strike></a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680331/"><strike>modules-integration-mysql57-jdk8/0/1</strike></a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645682/">modules-integration-mysql57-jdk8/0/2</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639370/"><strike>modules-integration-mysql57-jdk8/0/3</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266178/">modules-integration-mysql57-jdk8/0/4</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680327/">modules-integration-mysql57-jdk8/0/5</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639365/">modules-integration-mysql57-jdk8/0/6</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621356/">modules-integration-mysql57-jdk8/0/7</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639366/">modules-integration-mysql57-jdk8/0/8</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645678/"><strike>modules-integration-mysql57-jdk8/0/9</strike></a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600500/">modules-integration-mysql57-jdk8/0/10</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639362/"><strike>modules-integration-mysql57-jdk8/0/11</strike></a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645672/">modules-integration-mysql57-jdk8/0/12</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680322/">modules-integration-mysql57-jdk8/0/13</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685976/">modules-integration-mysql57-jdk8/0/14</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528466/">modules-integration-mysql57-jdk8/0/15</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528467/">modules-integration-mysql57-jdk8/0/16</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621352/"><strike>modules-integration-mysql57-jdk8/0/17</strike></a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680323/">modules-integration-mysql57-jdk8/0/18</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621354/">modules-integration-mysql57-jdk8/0/19</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266163/"><strike>modules-integration-mysql57-jdk8/0/20</strike></a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600494/">modules-integration-mysql57-jdk8/0/21</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600495/"><strike>modules-integration-mysql57-jdk8/0/22</strike></a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639355/"><strike>modules-integration-mysql57-jdk8/0/23</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685968/"><strike>modules-integration-mysql57-jdk8/0/24</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266176/">modules-integration-mysql57-jdk8/1/0</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621355/">modules-integration-mysql57-jdk8/1/1</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528471/">modules-integration-mysql57-jdk8/1/2</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680321/">modules-integration-mysql57-jdk8/1/3</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605278/"><strike>modules-integration-mysql57-jdk8/1/4</strike></a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605276/"><strike>modules-integration-mysql57-jdk8/1/5</strike></a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680319/">modules-integration-mysql57-jdk8/1/6</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266172/"><strike>modules-integration-mysql57-jdk8/1/7</strike></a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645670/"><strike>modules-integration-mysql57-jdk8/1/8</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685974/">modules-integration-mysql57-jdk8/1/9</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639348/"><strike>modules-integration-mysql57-jdk8/1/10</strike></a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680309/"><strike>modules-integration-mysql57-jdk8/1/11</strike></a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639350/">modules-integration-mysql57-jdk8/1/12</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528451/"><strike>modules-integration-mysql57-jdk8/1/13</strike></a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621338/">modules-integration-mysql57-jdk8/1/14</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528453/">modules-integration-mysql57-jdk8/1/15</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639351/">modules-integration-mysql57-jdk8/1/16</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645659/">modules-integration-mysql57-jdk8/1/17</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639352/"><strike>modules-integration-mysql57-jdk8/1/18</strike></a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645660/"><strike>modules-integration-mysql57-jdk8/1/19</strike></a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605267/">modules-integration-mysql57-jdk8/1/20</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528455/">modules-integration-mysql57-jdk8/1/21</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621340/">modules-integration-mysql57-jdk8/1/22</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680315/">modules-integration-mysql57-jdk8/1/23</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605268/"><strike>modules-integration-mysql57-jdk8/1/24</strike></a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605274/"><strike>modules-integration-mysql57-jdk8/2/0</strike></a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528460/">modules-integration-mysql57-jdk8/2/1</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685970/"><strike>modules-integration-mysql57-jdk8/2/2</strike></a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674220/"><strike>modules-integration-mysql57-jdk8/2/3</strike></a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600496/"><strike>modules-integration-mysql57-jdk8/2/4</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266166/"><strike>modules-integration-mysql57-jdk8/2/5</strike></a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605272/">modules-integration-mysql57-jdk8/2/6</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528459/">modules-integration-mysql57-jdk8/2/7</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621344/">modules-integration-mysql57-jdk8/2/8</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639357/">modules-integration-mysql57-jdk8/2/9</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528474/">modules-integration-mysql57-jdk8/2/10</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645679/"><strike>modules-integration-mysql57-jdk8/2/11</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685981/"><strike>modules-integration-mysql57-jdk8/2/12</strike></a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600505/"><strike>modules-integration-mysql57-jdk8/2/13</strike></a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605286/"><strike>modules-integration-mysql57-jdk8/2/14</strike></a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645680/"><strike>modules-integration-mysql57-jdk8/2/15</strike></a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674229/">modules-integration-mysql57-jdk8/2/16</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605287/"><strike>modules-integration-mysql57-jdk8/2/17</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685982/"><strike>modules-integration-mysql57-jdk8/2/18</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266184/">modules-integration-mysql57-jdk8/2/19</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621348/"><strike>modules-integration-mysql57-jdk8/2/20</strike></a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528462/">modules-integration-mysql57-jdk8/2/21</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680318/"><strike>modules-integration-mysql57-jdk8/2/22</strike></a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621349/">modules-integration-mysql57-jdk8/2/23</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645667/">modules-integration-mysql57-jdk8/2/24</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674217/"><strike>modules-integration-mysql57-jdk8/3/0</strike></a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674215/"><strike>modules-integration-mysql57-jdk8/3/1</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685960/">modules-integration-mysql57-jdk8/3/2</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266156/"><strike>modules-integration-mysql57-jdk8/3/3</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685959/"><strike>modules-integration-mysql57-jdk8/3/4</strike></a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605266/">modules-integration-mysql57-jdk8/3/5</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685961/"><strike>modules-integration-mysql57-jdk8/3/6</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266158/">modules-integration-mysql57-jdk8/3/7</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266157/">modules-integration-mysql57-jdk8/3/8</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600492/"><strike>modules-integration-mysql57-jdk8/3/9</strike></a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639354/">modules-integration-mysql57-jdk8/3/10</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674218/"><strike>modules-integration-mysql57-jdk8/3/11</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266162/"><strike>modules-integration-mysql57-jdk8/3/12</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685966/">modules-integration-mysql57-jdk8/3/13</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674219/"><strike>modules-integration-mysql57-jdk8/3/14</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266165/">modules-integration-mysql57-jdk8/3/15</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685967/"><strike>modules-integration-mysql57-jdk8/3/16</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266164/">modules-integration-mysql57-jdk8/3/17</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639356/">modules-integration-mysql57-jdk8/3/18</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685969/">modules-integration-mysql57-jdk8/3/19</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685965/">modules-integration-mysql57-jdk8/3/20</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605273/">modules-integration-mysql57-jdk8/3/21</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621346/">modules-integration-mysql57-jdk8/3/22</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621345/">modules-integration-mysql57-jdk8/3/23</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680316/">modules-integration-mysql57-jdk8/3/24</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685979/"><strike>modules-integration-mysql57-jdk8/4/0</strike></a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645677/"><strike>modules-integration-mysql57-jdk8/4/1</strike></a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621358/">modules-integration-mysql57-jdk8/4/2</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639364/">modules-integration-mysql57-jdk8/4/3</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528473/"><strike>modules-integration-mysql57-jdk8/4/4</strike></a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680328/">modules-integration-mysql57-jdk8/4/5</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266180/"><strike>modules-integration-mysql57-jdk8/4/6</strike></a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621359/">modules-integration-mysql57-jdk8/4/7</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639368/">modules-integration-mysql57-jdk8/4/8</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605285/"><strike>modules-integration-mysql57-jdk8/4/9</strike></a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680308/"><strike>modules-integration-mysql57-jdk8/4/10</strike></a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645662/">modules-integration-mysql57-jdk8/4/11</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685963/">modules-integration-mysql57-jdk8/4/12</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/645661/"><strike>modules-integration-mysql57-jdk8/4/13</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685962/">modules-integration-mysql57-jdk8/4/14</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266160/"><strike>modules-integration-mysql57-jdk8/4/15</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266161/">modules-integration-mysql57-jdk8/4/16</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621342/"><strike>modules-integration-mysql57-jdk8/4/17</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685964/"><strike>modules-integration-mysql57-jdk8/4/18</strike></a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639353/"><strike>modules-integration-mysql57-jdk8/4/19</strike></a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674216/">modules-integration-mysql57-jdk8/4/20</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266159/">modules-integration-mysql57-jdk8/4/21</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605284/">modules-integration-mysql57-jdk8/4/22</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266177/">modules-integration-mysql57-jdk8/4/23</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/528472/"><strike>modules-integration-mysql57-jdk8/4/24</strike></a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/621351/">modules-integration-mysql57-jdk8/5/0</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600497/">modules-integration-mysql57-jdk8/5/1</a></li></ul></li></ul></div><details><summary><strong>73 Successful Jobs:</strong></summary><ul/></details></details><h5>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/4476/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/600496//consoleText">modules-integration-mysql57-jdk8/2/4</a><pre>Build was aborted</pre><ol><li><div>com.liferay.headless.delivery.resource.v1_0.test.DocumentFolderResourceTest.testDeleteDocumentFolderMyRating - UNTESTED</div></li><li><div>com.liferay.headless.delivery.resource.v1_0.test.DocumentFolderResourceTest.testClientSerDesToDTO - UNTESTED</div></li><li><div>com.liferay.headless.delivery.resource.v1_0.test.DocumentFolderResourceTest.testClientSerDesToJSON - UNTESTED</div></li><li>...</li></ol></div></li><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/685979//consoleText">modules-integration-mysql57-jdk8/4/0</a><pre>Build was aborted</pre><ol><li><div>com.liferay.sharing.notifications.test.SharingUserNotificationTest.testNotificationMessage - UNTESTED</div></li><li><div>com.liferay.sharing.notifications.test.SharingUserNotificationTest.testAddUserNotification - UNTESTED</div></li><li><div>com.liferay.sharing.notifications.test.SharingUserNotificationTest.testAddUserNotificationWhenEmailNotificationsDisabled - UNTESTED</div></li><li>...</li></ol></div></li><li><div><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266160//consoleText">modules-integration-mysql57-jdk8/4/15</a><ol><li><details><summary><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266160/testReport/com.liferay.portal.workflow.metrics.rest.resource.v1_0.test/IndexResourceTest">com.liferay.portal.workflow.metrics.rest.resource.v1_0.test.IndexResourceTest</a> - <a href="https://testray.liferay.com/home/-/testray/case_results/2406336051/history">Failed 3 of last 24</a></summary><div><ol><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266160//testReport/com.liferay.portal.workflow.metrics.rest.resource.v1_0.test/IndexResourceTest/testGraphQLGetIndexesPage">testGraphQLGetIndexesPage</a></li></ol></div></details></li><li><details><summary><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266160/testReport/com.liferay.portal.workflow.metrics.rest.resource.v1_0.test/CalendarResourceTest">com.liferay.portal.workflow.metrics.rest.resource.v1_0.test.CalendarResourceTest</a> - <a href="https://testray.liferay.com/home/-/testray/case_results/2406336037/history">Failed 3 of last 24</a></summary><div><ol><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/266160//testReport/com.liferay.portal.workflow.metrics.rest.resource.v1_0.test/CalendarResourceTest/testGraphQLGetCalendarsPage">testGraphQLGetCalendarsPage</a></li></ol></div></details></li></ol></div></li></ol><hr/><details><summary><strong>Failures in common with <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at 2a91798016e3edcb4265da3039ad3157ba7ded7c:</strong></summary><ol><li><div><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639371//consoleText">modules-integration-mysql57-jdk8/0/0</a><ol><li><details><summary><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639371/testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest">com.liferay.portal.log.assertor.PortalLogAssertorTest</a></summary><div><ol><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/639371//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">testScanXMLLog</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680331//consoleText">modules-integration-mysql57-jdk8/0/1</a><ol><li><details><summary><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680331/testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest">com.liferay.portal.log.assertor.PortalLogAssertorTest</a></summary><div><ol><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/680331//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">testScanXMLLog</a></li></ol></div></details></li></ol></div></li><li>...</li></ol></details></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-35/jobs/test-portal-acceptance-pullrequest(master)/builds/4476/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-35/jobs/test-portal-acceptance-pullrequest(master)/builds/4476/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-35/jobs/test-portal-acceptance-pullrequest(master)/builds/4476/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/4476/">test-portal-acceptance-pullrequest(master)#4476</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/4476/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>upgrade<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-uniform/liferay-portal/pull/1338">liferay-uniform#1338</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2406446197">[master] ci:test:upgrade - lucasperj &gt; liferay-uniform - PR#1338 - 2023-08-14[19:21:56]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-35.liferay.com/job/publish-testray-report/9847/">publish-testray-report#9847</a><br/></div></div>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/65d3192f5ce94d93ebfd5019fffa6a025ca7ec07">65d3192f5ce94d93ebfd5019fffa6a025ca7ec07</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/lucasperj/liferay-portal/tree/LPS-177199">LPS-177199</a><br/>Branch GIT ID: <a href="https://github.com/lucasperj/liferay-portal/commit/adf0a06ee4f4dc740ef22e7d6424495ef33c849d">adf0a06ee4f4dc740ef22e7d6424495ef33c849d</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-3.liferay.com/job/test-portal-source-format/7182/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-source-format/builds/7182/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-3.liferay.com/job/test-portal-source-format/7182/">test-portal-source-format#7182</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-source-format/builds/7182/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-uniform/liferay-portal/pull/1338">liferay-uniform#1338</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2406531947">[master] ci:test:sf - lucasperj &gt; liferay-uniform - PR#1338 - 2023-08-15[09:07:51]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-3.liferay.com/job/publish-testray-report/3553/">publish-testray-report#3553</a><br/></div></div>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>upgrade</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest(master)/2727/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/2727/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:upgrade - 4 out of 8 jobs passed in 2 hours 41 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/65d3192f5ce94d93ebfd5019fffa6a025ca7ec07">65d3192f5ce94d93ebfd5019fffa6a025ca7ec07</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/65d3192f5ce94d93ebfd5019fffa6a025ca7ec07">65d3192f5ce94d93ebfd5019fffa6a025ca7ec07</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2406460273">EE Development Acceptance (master) - 4741 - 2023-08-15[00:38:03]</a></p><details><summary><strong>ci:test:upgrade - 4 out of 8 jobs PASSED</strong></summary><div><h4>4 Failed Jobs:</h4><ul><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest(master)/2727/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li><li>functional-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/251040/">functional-tomcat90-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/245004/">functional-tomcat90-mysql57-jdk8/0/1</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/265934/">functional-upgrade-tomcat90-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/254162/">functional-upgrade-tomcat90-mysql57-jdk8/0/1</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/241584/"><strike>functional-upgrade-tomcat90-mysql57-jdk8/0/2</strike></a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/267720/"><strike>functional-upgrade-tomcat90-mysql57-jdk8/0/3</strike></a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605900/"><strike>functional-upgrade-tomcat90-mysql57-jdk8/0/4</strike></a></li></ul></li></ul></div><details><summary><strong>4 Successful Jobs:</strong></summary><ul/></details></details><h5>For more details click <a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/2727/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/241584//consoleText">functional-upgrade-tomcat90-mysql57-jdk8/0/2</a><pre>A CI failure has occurred. QA Engineering has been notified. Please try again later or contact QA Engineering for an update.</pre><ol><li><div><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/241584//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_UpgradeManagerMbean_CheckMbeanSuccessfulUpgrade_">LocalFile.UpgradeManagerMbean#CheckMbeanSuccessfulUpgrade</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/2/LocalFile.UpgradeManagerMbean_CheckMbeanSuccessfulUpgrade/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/2/LocalFile.UpgradeManagerMbean_CheckMbeanSuccessfulUpgrade/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/2/LocalFile.UpgradeManagerMbean_CheckMbeanSuccessfulUpgrade/console.txt.gz">Poshi Console</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/2/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li><li><div><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/267720//consoleText">functional-upgrade-tomcat90-mysql57-jdk8/0/3</a><ol><li><div><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/267720//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_UpgradeManagerMbean_CheckMbeanWarningUpgrade_">LocalFile.UpgradeManagerMbean#CheckMbeanWarningUpgrade</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/3/LocalFile.UpgradeManagerMbean_CheckMbeanWarningUpgrade/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/3/LocalFile.UpgradeManagerMbean_CheckMbeanWarningUpgrade/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/3/LocalFile.UpgradeManagerMbean_CheckMbeanWarningUpgrade/console.txt.gz">Poshi Console</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/3/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li><li><div><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605900//consoleText">functional-upgrade-tomcat90-mysql57-jdk8/0/4</a><ol><li><div><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/605900//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_UpgradeManagerMbean_CheckMbeanWhenUpgradeHaveWarnAndError_">LocalFile.UpgradeManagerMbean#CheckMbeanWhenUpgradeHaveWarnAndError</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/4/LocalFile.UpgradeManagerMbean_CheckMbeanWhenUpgradeHaveWarnAndError/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/4/LocalFile.UpgradeManagerMbean_CheckMbeanWhenUpgradeHaveWarnAndError/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/4/LocalFile.UpgradeManagerMbean_CheckMbeanWhenUpgradeHaveWarnAndError/console.txt.gz">Poshi Console</a> - <a href="https://testray.liferay.com/reports/production/logs/2023-08/test-1-25/test-portal-acceptance-pullrequest(master)/2727/functional-upgrade-tomcat90-mysql57-jdk8/0/4/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/2727/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/2727/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/2727/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

vicnate5 commented:
@lucasperj Thanks for your patience iterating on this. I know that doing full new feature automation that requires new scripts takes more work than usual.

Can you check a few more things? Mostly around naming.
1. Using `testcase.url` to create the scenario where the portal keeps starting feels a bit hacky to me (though your solution to use the custom log message is great!). I am wondering if build-test.xml `start-app-server-tomcat-simple` could be used to do a partial startup?  --- If not, can you just use a more explicit property like `custom.startup.log.message`?
2. Rename `check-mbean-attributes` to `start-app-server-check-mbean-attributes` so we can make it clear that this isn't just checking the attributes but also starting the server.
3. Seems that you created `custom.catalina.opts` allows for future usage, however it also feels really tailored to our specific case (it also cleans up the mbean attribute file). Given that we don't have other usages and the properties are just for the mbean server, then can we change the property to `jmxremote.enabled`? Then hardcode the properties needed to enable it in ant. I feel like this better communicates what we are doing in the test.

Creating more properties adds more complexity, but it helps readability in the long term.

lucasperj commented:
@vicnate5  I decided to change and keep catalina.opts, if I really think it's necessary/good to remove it, lmk, the change is quick

1. I couldn't do it using tomcat-simple, and we still need to wait for the message in the log to make sure the process is registered on the mBean server, so I used the new property
2. Renamed
3. I decided to keep custom.catatalina.OPTS in case it is necessary to change to other tests in the future, either in our scope or other teams, I see it was worth having isolated.
I just added a step below so if it's testing our mBean, it deletes the output file.

In the end I just didn't want to be creating lots of new properties

vicnate5 commented:
@lucasperj

> In the end I just didn't want to be creating lots of new properties

1. I specifically stated "Creating more properties adds more complexity, but it helps readability in the long term."
2. If you remove the custom catalina.opts property now, you will actually reduce the number of new properties here, since you added the jmx one.
3. Having the jmx property handle all the setup simplifies what the test writer needs to know to add a test for jmx server, just add the jmx property (not also needing to know to add the custom catalina settings)

Please update.

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>upgrade</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest(master)/8636/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-38.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/8636/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:upgrade - 8 out of 8 jobs passed in 3 hours 11 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/9e8d478225f67b18d232e4a727e5dad63d3163ad">9e8d478225f67b18d232e4a727e5dad63d3163ad</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/0f42cd6b63a1464fcfce8f77324529cf03fcc23a">0f42cd6b63a1464fcfce8f77324529cf03fcc23a</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2406987276">EE Development Acceptance (master) - 4742 - 2023-08-15[20:37:54]</a></p><details><summary><strong>ci:test:upgrade - 8 out of 8 jobs PASSED</strong></summary><details><summary><strong>8 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest(master)/8636/">test-portal-acceptance-pullrequest(master)</a></li><li>functional-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/715238/">functional-tomcat90-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/715237/">functional-tomcat90-mysql57-jdk8/0/1</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/245983/">functional-upgrade-tomcat90-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/715240/">functional-upgrade-tomcat90-mysql57-jdk8/0/1</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/245981/">functional-upgrade-tomcat90-mysql57-jdk8/0/2</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/689632/">functional-upgrade-tomcat90-mysql57-jdk8/0/3</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/689631/">functional-upgrade-tomcat90-mysql57-jdk8/0/4</a></li></ul></li></ul></details></details><h5>For more details click <a href="https://test-1-38.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/8636/jenkins-report.html">here</a>.</h5></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-38/jobs/test-portal-acceptance-pullrequest(master)/builds/8636/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-38/jobs/test-portal-acceptance-pullrequest(master)/builds/8636/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-38/jobs/test-portal-acceptance-pullrequest(master)/builds/8636/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest(master)/8636/">test-portal-acceptance-pullrequest(master)#8636</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-38.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/8636/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>upgrade<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-uniform/liferay-portal/pull/1338">liferay-uniform#1338</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2407352440">[master] ci:test:upgrade - lucasperj &gt; liferay-uniform - PR#1338 - 2023-08-16[12:27:41]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-38.liferay.com/job/publish-testray-report/17815/">publish-testray-report#17815</a><br/></div></div>

lucasperj commented:
@vicnate5 , 
1. I mentioned that I didn't want to end to be creating lots of new properties before you've specifically stated "Creating more properties adds more complexity, but it helps readability in the long term.", so, sorry if I expressed myself badly, I was explaining my line of reasoning.
2. I was thinking of leaving both for possible future use, but I completely narrowed it down to the scope of the test.
3. I agree, but I also think it would be easier for different testers to read in the test file itself that custom catalina opts is needed to make it work, I believe the minority of testers goes down to the build logic level (build-test.xml ) and connect all the dots. Changed.

vicnate5 commented:
Pushed to BChan.
https://github.com/brianchandotcom/liferay-portal/pull/139327
